WooCommerce/Classes/ViewRelated/NotificationsBadgeController.swift

```diff
@@ -1,38 +1,57 @@
 import Foundation
 import UIKit
 
-final class NotificationsBadgeController {
-    /// Displays or Hides the Dot, depending on the new Badge Value
-    ///
-    func badgeCountWasUpdated(newValue: Int, tab: WooTab, in tabBar: UITabBar, tabIndex: Int) {
-        guard newValue > 0 else {
-            hideDotOn(tab, in: tabBar, tabIndex: tabIndex)
-            return
+enum NotificationBadgeType {
+    case primary
+    case secondary
+
+    var color: UIColor {
+        switch self {
+        case .primary:
+            return .primary
+        case .secondary:
+            return .accent
         }
+    }
+}
+
+/// Gathers the necessary data to update the badge on a tabbar tab
+struct NotificationsBadgeInput {
+    let shouldBeVisible: Bool
+    let type: NotificationBadgeType
+    let tab: WooTab
+    let tabBar: UITabBar
+    let tabIndex: Int
+}
 
-        showDotOn(tab, in: tabBar, tabIndex: tabIndex)
+final class NotificationsBadgeController {
+    /// Updates the tab badge depending on the provided input parameter
+    ///
+    func updateBadge(with input: NotificationsBadgeInput) {
+        input.shouldBeVisible ? showDotOn(with: input) : hideDotOn(with: input)
     }
 
     /// Shows the dot in the specified WooTab
     ///
-    func showDotOn(_ tab: WooTab, in tabBar: UITabBar, tabIndex: Int) {
-        hideDotOn(tab, in: tabBar, tabIndex: tabIndex)
-        let dot = PurpleDotView(frame: CGRect(x: DotConstants.xOffset,
-                                             y: DotConstants.yOffset,
-                                             width: DotConstants.diameter,
-                                             height: DotConstants.diameter),
-                               borderWidth: DotConstants.borderWidth)
-        dot.tag = dotTag(for: tab)
+    private func showDotOn(with input: NotificationsBadgeInput) {
+        hideDotOn(with: input)
+        let dot = DotView(frame: CGRect(x: DotConstants.xOffset,
+                                        y: DotConstants.yOffset,
+                                        width: DotConstants.diameter,
+                                        height: DotConstants.diameter),
```

Comment:
Actually, we only give a fixed diameter (this logic was there before this PR) when adding the dot to the tab bar view: we are passing the frame through the initializer. In case we had to use the `DotView` in other scenarios we could do the same, so it is not entirely hard coded.
In my opinion, since the tab bar view size does not change much pixelwise, I would leave the fixed diameter to make the code simpler without having to add a more complex logic, given that we would also have to calculate the offset to place it properly.